### PR TITLE
CIRC-949: Always set Checked out status after override renew.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
https://issues.folio.org/browse/CIRC-949 - Overriding failed renewal for aged to lost item, non-renewable loan policy does not change aged to lost status.

## Purpose
There are cases for which override renew is allowed. When a loan that matches multiple cases (item is not loanable and it is aged to lost) only logic for the first matched case is triggered and as a result an aged to lost item is still in aged to lost status + billing info is not cleared out.

## Approach
Introduce `overrideRenewLoan` method that is executed for all the cases and sets `Checked out` status regardless override case.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java [hunk 1]
```diff
@@ -255,6 +267,33 @@ public void cannotOverrideWhenOverrideDateBeforeCurrentDueDate() {
     assertThat(renewedLoan, hasValidationError(hasMessage("renewal would not change the due date")));
   }
 
+  @Test
+  public void nonLoanableAgedToLostItemShouldBeProperlyRenewed() {
```
Comment (marcjohnson-kint) [inline src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java:271]: What does properly renewed mean?

Comment (bohdan-suprun) [reply src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java:271]: It means status changed to checked out and billing date and time and billing flag are removed.

Comment (marcjohnson-kint) [reply src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java:271]: Could this test instead be renamed and expanded to be a general example of how it should be possible to renew a non loanable aged to lost item?

Comment (bohdan-suprun) [reply src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java:271]: Do you mean to create an API test instead?

Comment (marcjohnson-kint) [reply src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java:271]: > Do you mean to create an API test instead?

No, I'm glad this is a unit test. Are those changes expressed in the test the only changes made by this code?

Comment (bohdan-suprun) [reply src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java:271]: It seems no. This PR allows correct override of mixed override scenarios for a lost item: when item is lost and policy is not loanable/renewable, etc.

Comment (marcjohnson-kint) [reply src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java:271]: I'm not sure I follow, does that mean this PR involves changes that aren't described by this new test?

Comment (bohdan-suprun) [reply src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java:271]: No, this test is exactly the scenario described in jira issue.

**Review by marcjohnson-kint — APPROVED** [review]

File: src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java (hunk 1 above)
Comment (marcjohnson-kint) [reply src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java:271]: I think I'm a little lost. Given this fix is important, I'll resolve this conversation and approve.

**Pull request merged**